helix-term/src/commands.rs

```diff
@@ -2322,6 +2346,21 @@ fn buffer_picker(cx: &mut Context) {
                 None => SCRATCH_BUFFER_NAME,
             };
 
+            let icon = if let Some(icons) = icons {
+                // If there is no path, the icon will default to the common "file" icon
+                let path = match self.path.clone() {
+                    Some(p) => p,
+                    None => PathBuf::from("scratch"),
+                };
+                if let Some(icon) = icons.icon_from_path(&path) {
+                    Some(icon)
+                } else {
+                    None
+                }
+            } else {
+                None
+            };
```

Comment:
```suggestion
            let icon = icons
                .zip(self.path.as_ref())
                .and_then(|(icons, path)| icons.icon_from_path(path));
```

The pattern comes up again, I've marked them all (2) to keep track.